api/v1/routes/auth.py

```diff
@@ -0,0 +1,89 @@
+from fastapi import FastAPI, Depends, HTTPException, status, APIRouter
+from fastapi.security import OAuth2PasswordBearer, OAuth2PasswordRequestForm
+from sqlalchemy.orm import Session
+from pydantic import BaseModel
+from typing import Optional
+from ..models.user import User
+from ..models.profile import Profile
+from datetime import datetime, timedelta
+from api.v1.schemas.token import Token, LoginRequest
+from api.v1.schemas.auth import UserCreate
+from api.utils import json_response
+from api.db.database import get_db
+from api.utils.auth import authenticate_user, create_access_token, get_current_admin, get_current_user,hash_password,get_user
+
+
+from api.v1.models.org import Organization
+
+from api.v1.models.product import Product
+
+
+db = next(get_db())
+
+
+router = APIRouter()
+
+ACCESS_TOKEN_EXPIRE_MINUTES = 30
+
+@router.post("/login", response_model=Token,status_code=status.HTTP_200_OK)
+def login_for_access_token(login_request: LoginRequest, db: Session = Depends(get_db)):
+    user = authenticate_user(db, login_request.username, login_request.password)
+    if not user:
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="Incorrect username or password",
+            headers={"WWW-Authenticate": "Bearer"},
+        )
+    access_token_expires = timedelta(minutes=ACCESS_TOKEN_EXPIRE_MINUTES)
+    access_token = create_access_token(
+        data={"username": user.username}, expires_delta=access_token_expires
+    )
+    return {"access_token": access_token, "token_type": "bearer"}
+  
+@router.post("/register", response_model=UserCreate, status_code=status.HTTP_201_CREATED)
+def register_user(user: UserCreate, db: Session = Depends(get_db)):
```

Comment:
create a `scoped_session` for db, and import wherever it is needed, this method would cause unintended consequences if the endpoint is called with a `db` query parameter